packages/data/src/components/use-select/index.js

```diff
@@ -78,9 +80,23 @@ const renderQueue = createQueue();
  *
  * @return {Function}  A custom react hook.
  */
-export default function useSelect( _mapSelect, deps ) {
-	const mapSelect = useCallback( _mapSelect, deps );
+export default function useSelect( storeKey, _mapSelect, deps ) {
+	if ( arguments.length < 3 ) {
+		// Backwards compatible signature
+		[ storeKey, _mapSelect, deps ] = [
+			null,
+			arguments[ 0 ],
+			arguments[ 1 ],
+		];
+	}
```

Comment:
Can we instead just check the type of the first argument? If it's a function then it's `mapSelect`, if it's a string then it's `storeKey`?

Reply:
I agree that it might be handled differently.

In general, extending API surface for `useSelect` creates some challenges:
- how to update JSDoc comment to cover new usage to ensure that proper documentation is generated for developers
- how to announce new API in a way that it doesn't become confusing, is it clear when to use which form?